lightning-block-sync/src/rpc.rs

```diff
@@ -68,9 +87,11 @@ impl RpcClient {
 
 		let error = &response["error"];
 		if !error.is_null() {
-			// TODO: Examine error code for a more precise std::io::ErrorKind.
-			let message = error["message"].as_str().unwrap_or("unknown error");
-			return Err(std::io::Error::new(std::io::ErrorKind::Other, message));
+			let rpc_error = RpcError { 
+				code: error["code"].as_i64().unwrap_or(-1), 
```

Comment:
yeah I guess we could do that.  I'm not sure what case (if ever) it won't have a code and message.  I purposefully mapped it to RPC_MISC_ERROR since that's sort of what it is but happy to do it as Option's too if you think that makes more sense.

Reply:
maybe it'd be a little cleaner to make RpcError an enum with a struct variant and an "Unknown" variant when there's no code/message 🤷‍♂️

I guess you don't gain much, still need to do a match to find out error code

Reply:
Let's keep it as is. Like you said, it probably shouldn't ever happen and the code is used as a catch-all for when the RPC handling code throws an exception in Core.

Reply:
Ok, Matt had made a comment that we should probably just use a new RpcError enum that wraps the io Error. think it’s worth just doing that?

Reply:
That's a much larger change and is orthogonal, IMO, if you're just trying to include an extra piece of data. Changing the error type here means would make it inconsistent with `RestCleint`, and you'd ultimately need to wrap it again at the `BlockSourceError` level, anyhow, unless you want to change that as well. But that already is an enum.

Reply:
Yeah, that makes sense.  I was trying to avoid having to deal with a cascade of changes like that.  If we're okay with this solution then it would work for what we needed (access to bitcoind error code)